gegelatilib/include/importer/tpgGraphDotImporter.h

```diff
@@ -0,0 +1,400 @@
+#ifndef TPG_GRAPH_DOT_IMPORTER_H
+#define TPG_GRAPH_DOT_IMPORTER_H
+
+#include <stdexcept>
+#include <map>
+#include <string>
+#include <memory>
+#include <inttypes.h>
+#include <regex>
+#include <fstream>
+#include <cstdio>
+
+#include "learn/learningEnvironment.h"
+#include "tpg/tpgVertex.h"
+#include "tpg/tpgTeam.h"
+#include "tpg/tpgAction.h"
+#include "tpg/tpgEdge.h"
+#include "tpg/tpgGraph.h"
+
+/**
+*	\brief Maximum number of characters that can be read in a single line.
+*/
+#define MAX_READ_SIZE 1024
+
+namespace Importer {
+	/**
+	* \brief Class used to import a TPG graph from a dot file.
+	* It should be able to import a whole Learning agent object.
+	*/
+	class TPGGraphDotImporter {
+	protected:
+		/**
+		* \brief File in which the dot content is read during import.
+		*/
+		std::ifstream pFile;
+
+		/**
+		* \brief last Line read from file 
+		*
+		*	helps to pass a line that didn't match any regex to another function
+		*/
+		std::string lastLine;;
+
+		/**
+		* \brief The environment in which the TPGGRAPH will be built
+		*/
+		Environment env;
+
+		/**
+		* \brief TPGGraph imported from dot file.
+		*/
+		TPG::TPGGraph tpg;
```

Comment:
Because the tpg is an attribute of the importer, the TPG will be destroyed with its importer, and any reference retrieved on it will be invalid.

I think it would be easier to give a reference to an existing TPGGraph object to the importer constructor, and to fill it during import.